### PR TITLE
New Datasets Wizard

### LINKED ISSUE #54: Dataset form flow needs implementing
This includes adding datafiles.

### PR DESCRIPTION
**Pull request by govuklaurence**
Closes #54

### CODE REVIEW

File: app/controllers/datasets_controller.rb [hunk 1]
```diff
@@ -1,2 +1,107 @@
...
+
+  def new
+    @dataset = Dataset.new
+  end
+
+  def create
+    @dataset = Dataset.new(params.require(:dataset).permit(:id, :title, :summary, :description))
+    @dataset.organisation = current_user.primary_organisation
+
+    if @dataset.save
```
Comment (rossjones) [inline app/controllers/datasets_controller.rb]: We should probably slugify the title into the dataset.name field here as the end url will be /dataset/NAME

File: app/controllers/datasets_controller.rb [hunk 2]
```diff
@@ -1,2 +1,107 @@
...
+  def addfile
+    @dataset = current_dataset
+    @datafile = Datafile.new
+
+    if request.post?
+      file_params = params.require(:datafile).permit(:url, :name)
+      @datafile = Datafile.new(file_params)
+      @datafile.dataset = @dataset
+
+      if @datafile.save
```
Comment (rossjones) [inline app/controllers/datasets_controller.rb]: When we save a datafile, I guess this is for the model, we need to set the start_date,end_date fields based on any date data we got from the form (like month/year, or q/year).

File: app/models/dataset.rb [hunk 3]
```diff
@@ -1,3 +1,19 @@
 class Dataset < ApplicationRecord
   belongs_to :organisation
+  has_many :datafiles
+
+  validates :title, presence: true, uniqueness: true
+  validates :summary, presence: true
```
Comment (rossjones) [inline app/models/dataset.rb]: Before we publish, we also need to validate the presence of:
1. A licence  (licence or licence_other).
2. A frequency (from a fixed list).
3. At least one datafile

File: app/controllers/datasets_controller.rb (hunk 1 above)
Comment (govuklaurence) [inline app/controllers/datasets_controller.rb]: agreed. this was just getting it working ASAP :) let's spin it out into an issue

File: app/controllers/datasets_controller.rb (hunk 2 above)
Comment (govuklaurence) [inline app/controllers/datasets_controller.rb]: Understood -- I'll include this in a seperate issue with the changes to frequency

File: app/models/dataset.rb (hunk 3 above)
Comment (govuklaurence) [inline app/models/dataset.rb]: Spinning out

Comment (govuklaurence) [inline app/models/dataset.rb]: #82

File: app/controllers/datasets_controller.rb (hunk 2 above)
Comment (govuklaurence) [inline app/controllers/datasets_controller.rb]: #81

File: app/controllers/datasets_controller.rb (hunk 1 above)
Comment (govuklaurence) [inline app/controllers/datasets_controller.rb]: #80

**Pull request merged**